internal/validitywindow/handler_test.go

```diff
@@ -15,6 +15,9 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// Use a fixed seed for deterministic test results
+var seed = rand.New(rand.NewSource(42)) //nolint:gosec
```

Comment:
```suggestion
var seed = rand.New(rand.NewSource(0)) //nolint:gosec
```

why 42?

Reply:
I don't think this will do what you want. If tests are run in a non-deterministic order or if you run a different subset of tests, then the order in which we read from `seed` will result in different random values.

It seems we only use this in `generateBlockChain(...)`, what if we create and use a deterministic seed within that function, so that generating a chain of the same length is deterministic?

Reply:
Great reason